### PR TITLE
feat:[IE-410]: Update external secrets for verification-svc

### PR DESCRIPTION
**Pull request by Vbhatia20**
## Describe your changes
External secrets support for verification-svc

Screenshot 2023-10-27 at 6 51 06 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53725)

### CODE REVIEW

File: 270-verification/chart/values.yaml [hunk 1]
```diff
@@ -33,6 +33,24 @@ global:
       userKey: ""
       passwordKey: ""
       extraArgs: ""
+      secrets:
+        kubernetesSecrets:
+          - secretName: ""
+            keys:
+              MONGO_USER: ""
+              MONGO_PASSWORD": ""
```
Comment (saitejavn-harness) [inline 270-verification/chart/values.yaml:41]: typo: Remove `"` at the end

File: 270-verification/chart/values.yaml [hunk 2]
```diff
@@ -33,6 +33,24 @@ global:
...
+        secretManagement:
+          externalSecretsOperator:
+            - secretStore:
+                name: ""
+                kind: ""
+              remoteKeys:
+                MONGO_USER:
+                  name: ""
+                  property: ""
+                MONGO_PASSWORD":
```
Comment (saitejavn-harness) [inline 270-verification/chart/values.yaml:51]: typo: Remove `"` at the end

File: 270-verification/chart/values.yaml [hunk 3]
```diff
@@ -122,6 +140,32 @@ serviceAccount:
...
+  secretName: ""
+  userKey: ""
+  passwordKey: ""
+  extraArgs: ""
+  secrets:
+    kubernetesSecrets:
+      - secretName: ""
+        keys:
+          MONGO_USER: ""
+          MONGO_PASSWORD": ""
```
Comment (saitejavn-harness) [inline 270-verification/chart/values.yaml:155]: typo: Remove `"` at the end

File: 270-verification/chart/values.yaml [hunk 4]
```diff
@@ -122,6 +140,32 @@ serviceAccount:
...
+    secretManagement:
+      externalSecretsOperator:
+        - secretStore:
+            name: ""
+            kind: ""
+          remoteKeys:
+            MONGO_USER:
+              name: ""
+              property: ""
+            MONGO_PASSWORD":
```
Comment (saitejavn-harness) [inline 270-verification/chart/values.yaml:165]: typo: Remove `"` at the end

File: 270-verification/chart/templates/deployment.yaml [hunk 5]
```diff
@@ -71,11 +71,11 @@ spec:
           - configMapRef:
               name: verification-svc
           env:
-            {{- include "harnesscommon.dbconnection.mongoEnv" . | indent 12 }}
+            {{- include "harnesscommon.dbconnectionv2.mongoEnv" (dict "ctx" $) | indent 12 }}
```
Comment (saitejavn-harness) [inline 270-verification/chart/templates/deployment.yaml:74]: Missing env for `VERIFICATION_SERVICE_SECRET`

**Review by saitejavn-harness — COMMENTED** [review]
Can we externalize this? https://github.com/harness/harness-core/blob/d48d878ed93f6c501cfbe6a9e84d3af9c658d1e6/270-verification/chart/templates/deployment.yaml#L132

**Review by saitejavn-harness — DISMISSED** [review]
change base branch to develop

**Comment by Vbhatia20** [comment]
trigger messagecheck

**Comment by Vbhatia20** [comment]
trigger uts

**Comment by Vbhatia20** [comment]
trigger gitleaks

**Comment by Vbhatia20** [comment]
trigger messagecheck

**Comment by Vbhatia20** [comment]
trigger messagecheck

**Comment by Vbhatia20** [comment]
trigger gitleaks

**Comment by Vbhatia20** [comment]
trigger uts

**Pull request closed without merging**